docs/updating/migration-to-cdn/react.md

```diff
@@ -0,0 +1,461 @@
+	} );
+
+	if ( cloud.status === 'loading' ) {
+		return <div>Loading...</div>;
+	}
+
+	if ( cloud.status === 'error' ) {
+		console.error( cloud.error );
+
+		return <div>Error!</div>;
+	}
+
+	// Pick the CKEditor 5 plugins you want to use.
+	const {
+		ClassicEditor, Bold, Essentials, Italic,
+		Mention, Paragraph, SlashCommand, Undo
+	} = cloud.CKEditor;
+
+	const { SlashCommand } = cloud.CKEditorPremiumFeatures;
+
+	return (
+		<CKEditor
+			editor={ ClassicEditor }
+			config={ {
+				toolbar: {
+					items: [ 'undo', 'redo', '|', 'bold', 'italic' ],
+				},
+				plugins: [
+					Bold,
+					Essentials,
+					Italic,
+					Mention,
+					Paragraph,
+					SlashCommand,
+					Undo
+				],
+				licenseKey: '<YOUR_LICENSE_KEY>', // Or 'GPL'.
+				initialData: '<p>Hello from CKEditor 5 in React!</p>',
+			} }
+		/>
+	);
+}
+```
+
+#### Step 3 (Optional): Migrate the CKEditor&nbsp;5 React testing suite
+
+If you have any tests that use CKEditor&nbsp;5 objects, you need to update them to use the  `loadCKEditorCloud` function. Here is an example of migrating a test that uses the `ClassicEditor` object:
+
+**Before:**
+
+```javascript
+import { ClassicEditor, ... } from 'ckeditor5';
+import { AIAdapter, ... } from 'ckeditor5-premium-features';
+
+it( 'ClassicEditor test', () => {
+	render(
+		<CKEditor
+			editor={ ClassicEditor }
+			// ... other props
+		/>
+	);
+} );
+```
+
+**After:**
+
+```javascript
+// It may be counterintuitive that in tests you need to use `loadCKEditorCloud` instead of `useCKEditorCloud`.
+// The reason for this is that `useCKEditorCloud` is a React hook and can only be used in React components,
+// while tests are typically written as functions in testing suites. Therefore, in tests, you should use
+// the `loadCKEditorCloud` function to load CKEditor 5 from the CKEditor Cloud and obtain the necessary
+// CKEditor 5 objects. This allows you to properly test your CKEditor 5 integration without any issues.
+
+import { loadCKEditorCloud } from '@ckeditor/ckeditor5-react';
+
+let cloud;
+
+beforeEach( async () => {
+	cloud = await loadCKEditorCloud( {
+		version: '{@var ckeditor5-version}',
+	} );
+} );
+
+it( 'ClassicEditor test', () => {
+	const { ClassicEditor } = cloud.CKEditor;
+
+	render(
+		<CKEditor
+			editor={ ClassicEditor }
+			// ... other props
+		/>
+	);
+
+	// Rest of your test.
+} );
+```
+
+#### Step 4 (Optional): Clean up the document head entries before each test
+
+The `useCKEditorCloud` hook under the hood injects the CKEditor&nbsp;5 scripts and styles into your document head. If you are using a testing suite that does not Clean up the document head entries before each test, you may need to do it manually. This is important because the `useCKEditorCloud` hook might reuse the same head entries for each test, which can lead to skipping `loading` state and directly going to the `success` state. It may cause some tests that rely on the `loading` state to fail.
```

Comment:
```suggestion
The `useCKEditorCloud` hook under the hood injects the CKEditor&nbsp;5 scripts and styles into your document head. If you use a testing suite that does not Clean up the document head entries before each test, you may need to do it manually. This is important because the `useCKEditorCloud` hook might reuse the same head entries for each test, which can lead to skipping the `loading` state and directly going to the `success` state. It may cause some tests that rely on the `loading` state to fail.
```